packages/core/src/connectors/alipay/index.ts

```diff
@@ -0,0 +1,245 @@
+/**
+ * The Implementation of OpenID Connect of Alipay Web Open Platform.
+ * https://opendocs.alipay.com/support/01rg6h
+ * https://opendocs.alipay.com/open/263/105808
+ * https://opendocs.alipay.com/open/01emu5
+ */
+import * as crypto from 'crypto';
+import { existsSync, readFileSync } from 'fs';
+import path from 'path';
+
+import got, { RequestError as GotRequestError } from 'got';
+import * as iconv from 'iconv-lite';
+import { stringify } from 'query-string';
+import snakeCaseKeys from 'snakecase-keys';
+import { z } from 'zod';
+
+import assertThat from '@/utils/assert-that';
+import { redirectUriRegEx } from '@/utils/regex';
+
+import {
+  ConnectorError,
+  ConnectorErrorCodes,
+  ConnectorMetadata,
+  ConnectorType,
+  GetAccessToken,
+  GetAuthorizationUri,
+  GetUserInfo,
+  ValidateConfig,
+} from '../types';
+import { getConnectorConfig, getConnectorRequestTimeout, getFormattedDate } from '../utilities';
+import {
+  alipayEndpoint,
+  authorizationEndpoint,
+  methodForAccessToken,
+  methodForUserInfo,
+  scope,
+} from './constant';
+
+const ALIPAY_ALGORITHM_MAPPING = {
```

Comment:
Can this be an enum as `CHAR_SET`?

Reply:
We are very careful with upper letters, better use:
```
const alipayAlgorithmMapping
```